### ISSUE TITLE
Missing characters when fast typing in IE (Internet Explorer 11), poor performance on IE and Firefox. Related to updateImmediately

### ISSUE THREAD

**Issue by dahei** [rojobuffalo/formsy-material-ui#214]
The performance of FormsyText in InternetExplorer 11 (and also Firefox) is not very pleasant. When using a default FormsyText component and keep pressing a letter on the keyboard, the  characters appear one after another with distinct breaks between each character. A normal input (and also the original (un)controlled TextField component from material-ui) has no such lag and always feels responsive.

I have recorded this behaviour here:

https://youtu.be/u9tlqf2QFxw

To avoid this lag I use updateImmediately={true}, which results in a much better performance. Unfortunately this leads to the problem, that while typing some characters are "overwritten" by the next one. It looks like the just typed character gets erased and instantly replaced by the next typed symbol.  I've also recorded that:

https://youtu.be/bzVqjfsmUKY?t=5

As you can see, the 'd' gets "replaced" by an 'f'. This is real problem for me, as my customers (using the Internet Explorer 11) are not happy neither with the performance (w/o updateImmediately) nor the missing characters.

As the wrapped original TextField of material-ui doesn't have this problem, FormsyText seems to be root of the problem.

There are two solutions in my mind:

1.) fix performance lags, so IE and Firefox show decent user experience (best solution)
or
2.) fix updateImmediately, so that no characters are missing or replaced by others.

These tests were recorded on Windows 10, Internet Explorer 11 and formsy-material-ui@0.6.0

It is the webpack-example App, which is part of the repo.

**Comment by dahei** [comment]
I've updated to the latest formsy-material-ui@0.6.3 and it's peer dependencies (react@^15.6.1 etc.) Fortunately, the described issue is gone and `FormsyText` with `updateImmediately` works as expected on Internet Explorer 11.

React 15.6.x solved a lot of IE 11 related bugs (https://facebook.github.io/react/blog/2017/06/13/react-v15.6.0.html) - maybe that solved that problem as well.

tl;dr: This issue can be closed.
